src/charm.py

```diff
@@ -64,6 +71,78 @@ def __init__(self, *args):
         self.framework.observe(self.on.update_status, self._on_update_status)
         self._storage_path = self.meta.storages["pgdata"].location
 
+    def _on_old_db_relation_changed(self, event: RelationChangedEvent) -> None:
+        """Handle the legacy db relation changed event."""
+        if not self.unit.is_leader():
+            return
+
+        self.unit.status = MaintenanceStatus("Setting up db relation")
+        logger.warning("DEPRECATION WARNING - `db` is a legacy interface")
+
+        unit_relation_databag = event.relation.data[self.unit]
+
+        if unit_relation_databag.get("user"):
+            # Test if relation data is already set
+            # and avoid overwriting it
+            logger.warning("Data for db already set.")
+            self.unit.status = ActiveStatus()
+            return
+
+        hostname = self._get_hostname_from_unit(self.unit.name.replace("/", "-"))
+        # Use connstring (https://pypi.org/project/pgconnstr/).
+        connection = connect_to_database(
+            "postgres", "postgres", hostname, self._get_postgres_password()
+        )
+        logger.info(f"Connected to PostgreSQL: {connection}")
+
+        user = f"relation_id_{event.relation.id}_{event.app.name.replace('-', '_')}"
+        password = self._new_password()
+        logger.error(str(event.relation.data))
+        database = event.relation.data[event.app].get("database")
+        if not database:
```

Comment:
Would it be worth adding some logs to tell the user about this? If not, could you please add a comment describing it to a reader?